shell/components/mic-input.js

```diff
@@ -0,0 +1,141 @@
+/*
+@license
+Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+
+import Xen from './xen/xen.js';
+import Icons from './icons.css.js';
+
+const log = Xen.logFactory('MicInput', 'blue');
+
+class MicInput extends Xen.Base {
```

Comment:
Brief docs on how this works / how to use it in other elements would be helpful.